lib/runners/processor/clang_tidy.rb

```diff
@@ -0,0 +1,70 @@
+module Runners
+  class Processor::ClangTidy < Processor
+    Schema = StrongJSON.new do
+      let :runner_config, Schema::BaseConfig.base.update_fields { |fields|
+        fields.merge!({})
+      }
+
+      let :issue, object(
+        severity: string,
+      )
+    end
+
+    register_config_schema(name: :clang_tidy, schema: Schema.runner_config)
+
+    FILE_EXTENSIONS = [".c", ".cc", ".cpp", ".c++", ".cp", ".cxx"].freeze
+
+    def analyze(changes)
+      run_analyzer(changes.changed_paths)
+    end
+
+    private
+
+    def analyzer_bin
+      "clang-tidy"
+    end
+
+    def run_analyzer(changed_paths)
+      issues = []
+
+      changed_paths
+        .select { |path| FILE_EXTENSIONS.include?(path.extname.downcase) }
+        .map{ |path| relative_path(working_dir / path, from: current_dir) }
+        .each do |path|
+        stdout, stderr = capture3!(analyzer_bin, path.to_s, "--")
+        ret = construct_result(stdout)
+        issues.push(*ret)
+      end
+
+      Results::Success.new(guid: guid, analyzer: analyzer).tap do |result|
+        result.add_issue(*issues)
+      end
+    end
+
+    def construct_result(stdout)
+      issues = []
+
+      stdout.each_line do |line|
+        # group: 1      2      3         4           5          6
+        #        <path>:<line>:<column>: <severity>: <message> [<id>]
+        match = line.strip.match(/^(?<path>.+):(?<line>\d+):(?<column>\d+): (?<severity>[^:]+): (?<message>.+) \[(?<id>[^\[]+)\]$/)
+        if match
+          start_line = Integer(match[:line])
+          start_column = Integer(match[:column])
+          issues << Issue.new(
+            path: relative_path(match[:path]),
+            location: Location.new(start_line: start_line, start_column: start_column, end_line: start_line, end_column: start_column),
```

Comment:
The following seems more natural to me. But currently, we cannot do it.

```suggestion
            location: Location.new(start_line: start_line, start_column: start_column),
```

I suggested allowing this form above via #1257.

Reply:
That's exactly what I want. Thank you.

Reply:
@suetsuguk I've merged #1257. 💪

Reply:
Thanks to #1257, I have removed end_line and end_column from my code at 023c68d. Looks nice. Thank you for your great work.